lib/explorer/polars_backend/data_frame.ex

```diff
@@ -748,6 +748,11 @@ defmodule Explorer.PolarsBackend.DataFrame do
     |> Shared.create_dataframe()
   end
 
+  @impl true
+  def explode(df, columns) do
+    Shared.apply_dataframe(df, df, :df_explode, [columns])
```

Comment:
I think the callback could receive the `out_df` with the  "correct" dtype. It's OK to have the first version without it, but I think we can say what dtypes the out DF is going to have.

Reply:
@philss done it in [6622caa](https://github.com/elixir-explorer/explorer/pull/751/commits/6622caac96a40c4e08852fe4d6c07c38bfd291a4)

I also found a bug when aggregating the series elements into lists while implementing this. The commit above should also fix that bug (I've added a test for it).

Reply:
@costaraphael excellent! :sparkles:
thanks!